DEV-3/DEV-3/TeamSearcherWithMaxProductivity.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DEV_3
+{
+    /// <summary>
+    /// This class can find the best team for money that costumer has.
+    /// </summary>
+    class TeamSearcherWithMaxProductivity : TeamSearcher
+    {
+        /// <summary>
+        /// The class constructor initializes fields.
+        /// </summary>
+        /// <param name="amountOfMoney">Amount of money that costumer has</param>
+        /// <param name="productivity">Productivity that costumer requires</param>
+        public TeamSearcherWithMaxProductivity(int amountOfMoney, int productivity)
+        {
+            AmountOfMoney = amountOfMoney;
+            Productivity = productivity;
+        }
+
+        /// <summary>
+        /// This method finds and returns the best team for money that costumer has.
+        /// </summary>
+        /// <param name="availableEmployees"></param>
+        /// <returns></returns>
+        internal override  List<Employee> Choose(List<Employee> availableEmployees)
+        {
+        leads:
+            while (AmountOfMoney >= lead.Salary)
+            {
+                foreach (var employee in availableEmployees)
+                {
+                    if (employee is Lead)
+                    {
+                        team.Add(employee);
+                        availableEmployees.Remove(employee);
+                        AmountOfMoney -= lead.Salary;
+                        currentProductivity += lead.Productivity;
+                        goto leads;
```

Comment:
goto? is it OOP?

Reply:
Can I somehow break my while loop in foreach loop? Or I need to fully rebuild logic?